addons/pos_self_order/models/product_product.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from typing import List, Dict, Optional
+
+from odoo import models
+
+from odoo.addons.point_of_sale.models.pos_config import PosConfig
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _get_name(self) -> str:
+        """
+        Returns the name of the product without the code.
+        ex: product_sudo.display_name is '[FURN_7888] Desk Stand with Screen (Red)'
+        :return: 'Desk Stand with Screen (Red)' (we remove the [FURN_7888] part)
+        """
+        self.ensure_one()
+        # product_sudo.code would be 'FURN_7888', but we also need to remove the brackets
+        # and the space after the brackets (hence the +3)
+        return self.code and self.display_name[len(self.code) + 3 :] or self.display_name
+
+    def _filter_applicable_attributes(self, attributes_by_ptal_id: Dict) -> List[Dict]:
+        """
+        The attributes_by_ptal_id is a dictionary that contains all the attributes that have
+        [('create_variant', '=', 'no_variant')]
+        This method filters out the attributes that are not applicable to the product in self
+        """
+        self.ensure_one()
+        return [
+            attributes_by_ptal_id[id]
+            for id in self.attribute_line_ids.ids
+            if id in attributes_by_ptal_id and attributes_by_ptal_id[id] is not None
```

Comment:
```
attributes_by_ptal_id[id]
```
if `id` is not a key of `attributes_by_ptal_id`, that key access will crash. You maybe want `attributes_by_ptal_id.get(id)` if you're expecting potential `None` return value.